### ISSUE TITLE
Hanging after each command with zsh

### ISSUE THREAD

**Issue by arlyon** [cantino/mcfly#317]
Hi

I am using this with zinit with the setup commands in the readme, and have discovered that, with a db size of around 25,000, every command hangs for about 3-5 seconds. I can't really figure out _why_, since it running `mcfly add` directly runs in a few milliseconds, but I have had to disable the plugin for now.

Is this known behaviour?

Alex

**Comment by cantino** [comment]
No, I’ve never seen this behavior. Is it a reasonably fast computer?

**Comment by arlyon** [comment]
5850U 16 threads so pretty snappy. Is there a way to enable verbose logging for the zsh integration (perhaps exposing the rust logging / tracing?). I'll attempt to get some logs with timestamps.

**Comment by cantino** [comment]
You could make a temporary file containing the init script and add echo statements to see where it's hanging, perhaps? See this comment: https://github.com/cantino/mcfly/issues/313#issuecomment-1352438906
